extensions/default/src/CloudServerAPI/GoogleCloudServerAPI/GoogleCloudServerAPI.ts

```diff
@@ -0,0 +1,138 @@
+import { Types } from '@ohif/core';
+
+async function _doRequest(urlStr, config = {}, params = {}): Promise {
+  const url = new URL(urlStr);
+  let data = null;
+  url.search = new URLSearchParams(params);
+
+  try {
+    const response = await fetch(url, config);
+    try {
+      data = await response.json();
+    } catch (err) {}
+    if (response.status >= 200 && response.status < 300 && data != null) {
+      if (data.nextPageToken != null) {
+        params.pageToken = data.nextPageToken;
+        const subPage = await _doRequest(urlStr, config, params);
+        for (const key in data) {
+          if (data.hasOwnProperty(key)) {
+            data[key] = data[key].concat(subPage.data[key]);
+          }
+        }
+      }
+      return {
+        isError: false,
+        status: response.status,
+        data,
+      };
+    } else {
+      return {
+        isError: true,
+        status: response.status,
+        message: (data && data.error && data.error.message) || 'Unknown error',
+      };
+    }
+  } catch (err) {
+    if (data && data.error) {
+      return {
+        isError: true,
+        status: err.status,
+        message: err.response.data.error.message || 'Unspecified error',
+      };
+    }
+    return {
+      isError: true,
+      message: (err && err.message) || 'Oops! Something went wrong',
+    };
+  }
+}
+
+class GoogleCloudServerItem implements Types.BaseCloudServerAPIItem {
```

Comment:
Here are the implementations of the CloudServerAPI interfaces for Google. Again, very rough. There is no error checking. But I hope the idea is clear.